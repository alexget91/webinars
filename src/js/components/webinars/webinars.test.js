import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import Webinars from "./webinars";

it(`Webinars correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<MemoryRouter>
      <Webinars
        cards={[{
          id: 1,
          title: `title`,
          description: `description`,
          imageSrc: `imageSrc`,
        }]}
        currentPage={1}
        onCardAddSubmit={jest.fn()}
      />
    </MemoryRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
