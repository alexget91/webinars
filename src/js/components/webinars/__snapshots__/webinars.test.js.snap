// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webinars correctly renders after relaunch 1`] = `
Array [
  <section
    className="main-header"
  >
    <h1
      className="main-header__title"
    >
      Webinars
    </h1>
    <p
      className="main-header__text"
    >
      Here you can register and take part in educational webinars conducted by the best digital marketing experts.
    </p>
    <button
      className="btn"
      onClick={[Function]}
    >
      Add new
    </button>
  </section>,
  <section
    className="catalog"
  >
    <div
      className="catalog__content"
    >
      <article
        className="card catalog__card"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="title"
            className="card__image"
            src="imageSrc"
          />
        </a>
        <div
          className="card__content"
        >
          <h2
            className="card__title"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              title
            </a>
          </h2>
          <p
            className="card__text"
          >
            description
          </p>
          <div
            className="card__footer"
          />
        </div>
      </article>
      <div
        className="card card--double catalog__card catalog__card--double"
      >
        <a
          className="card__double-inner"
          href="/"
          onClick={[Function]}
        >
          <div
            className="catalog__mobile-hidden"
          >
            <img
              alt="Banner"
              src="test-file-stub"
            />
          </div>
          <div
            className="catalog__desktop-hidden"
          >
            <img
              alt="Banner"
              src="test-file-stub"
            />
          </div>
        </a>
      </div>
    </div>
    <div
      className="pagination catalog__pagination"
    >
      <a
        className="pagination__link active"
        href="/?page=1"
        onClick={[Function]}
      >
        1
      </a>
    </div>
  </section>,
]
`;
